docs/source/concepts/pipeline_schema/index.rst

```diff
@@ -983,3 +986,247 @@ We expect that these constraints will typically be enforced during this step.
+
+**Examples:**
+
+- The `Unicorn <https://dl.acm.org/doi/abs/10.1145/3588938>`_ model contains automatic schema alignment.
+
+.. _records:
+
+Records
+^^^^^^^
+
+**Interpretation:**
+The records to link (from all datasets) in one big table.
+
+**Specification:**
+A file in a tabular format.
+The file may have any number of columns,
+but one of them must be called “Input Record ID” and it must have unique values.
+
+.. note::
+
+   In the future, we should add to this specification that every value in the Input Record ID column
+   should exist in one of the input datasets.
+   EasyLink currently doesn't support this.
+
+**Example:**
+
+.. list-table:: 
+   :header-rows: 1
+
+   * - Input Record ID
+     - First
+     - Last
+     - Address
+   * - input_file_1
+     - Vicki
+     - Simmons
+     - 123 Main St. Apt C, Anytown WA 99999
+   * - input_file_2
+     - Gerald
+     - Allen
+     - 456 Other Drive, Anytown WA, 99999
+   * - reference_file_1
+     - Victoria
+     - Simmons
+     - 123 Main St., Anytown WA 99999
+   * - reference_file_2
+     - Gerry
+     - Allen
+     - *N/A*
+
+Blocking and filtering
+^^^^^^^^^^^^^^^^^^^^^^
+
+**Interpretation:**
+Breaking the linkage problem up into pieces that can be tackled separately,
+and selecting which pairs of records to consider in each piece,
+in order to reduce the size of the task and therefore the computation required.
+
+Traditional blocking, where a blocking "key" is assigned to each record,
+implements this step by splitting the records into blocks (disjoint subsets)
+by their blocking key and enumerating all possible pairs within each block.
+
+More advanced techniques may instead create just *one* block (with all records),
+and select only some pairs within that block rather than every possible pair.
+
+Techniques focused on or configured for linkage *between* datasets can avoid enumerating
+pairs of records within the same dataset.
+
+This step corresponds to "indexing" in `Christen (2012) <https://link.springer.com/book/10.1007/978-3-642-31164-2>`_.
+
+**Examples:**
+
+- In Splink, using a single blocking rule would be traditional blocking as described above:
+  a separate block for each value of date of birth, for instance.
+  Multiple blocking rules in Splink are OR'd together, creating overlapping blocks.
+  In EasyLink, this could be represented as putting all records in a single block but only
+  enumerating the pairs matching at least one of the blocking rule conditions.
+
+Blocks
+^^^^^^
+
+**Interpretation:**
+Separate pieces of the linkage task that can be tackled separately,
+along with the pairs of records to consider in each.
+
+**Specification:**
+A directory containing any number of subdirectories.
+Each subdirectory must contain two files, each in tabular format: records and pairs.
+
+Each records file must follow the specification for :ref:`Records`.
+
+Each pairs file must contain two columns, "Left Record ID" and "Right Record ID".
```

Comment:
We might want a shorthand for "all pairs"; it seems inefficient to write them all out

Reply:
Weird coincidence that this article crossed my inbox today https://towardsdatascience.com/efficient-graph-storage-for-entity-resolution-using-clique-based-compression/

Reply:
@aflaxman My thinking is that if you have to do some computation and store resulting probabilities for each pair (which seems unavoidable with a pairwise method), then storing the list of pairs is never the bottleneck. Do you think there is a case I'm not thinking of?

Reply:
Maybe there are settings where you don't need to store low probabilities, so you can avoid writing to disk.  We can circle back to this if/when it seems like it might speed things up.